server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1476,6 +1493,7 @@ private static class IndexMetadataDiff implements Diff<IndexMetadata> {
         private final IndexMetadataStats stats;
         private final Double indexWriteLoadForecast;
         private final Long shardSizeInBytesForecast;
+        private final Diff<Map<String, Set<String>>> fieldsForModels;
```

Comment:
How do we usually go about testing `IndexMetadataDiff`?

Reply:
I didn't look into the testing side TBH. I guess we could check [here](https://github.com/elastic/elasticsearch/blob/1f3126b47b50f93a5b6c595cc34c1bd7a94168b3/server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterStateDiffIT.java#L573) for mutating the `IndexMetadata` so it includes changes to the wider cluster state diffs, which are checked here.

We can look deeper into that 👍

Reply:
Updated tests in `ClusterStateDiffIT` to cover this